Adaptive smoothing updates

kirstybayliss opened the pull request:
This PR:
* adds tests for the mbt adaptive smoothing function and the wkf adaptive smoothing function
* adds functionality to the mbt adaptive smoothing such that it can be used with h3 cells and with a specified maximum distance
* improves the wkf adaptive smoothing function and h3/zones.py to use only specified model polygons and to require a maximum distance (_**NB**_: _small change to config for wkf function, smoothing parameters should be under [smoothing] header in toml)
* small changes to wkf_adaptive_smoothing to make compatible with the output of set_h3_to_zones and therefore more easily comparable to gaussian smoothing

kirstybayliss commented:
The last test in adaptive_smoothing_wkf_test (test_adap_smoothing_wkf) passes for me locally, but the subprocess call doesn't complete correctly when running the tests here. The other tests in adaptive_smoothing_wkf_test test all of the other components of the wkf function, it seems that only the subprocess call is the problem here.

mmpagani commented on openquake/wkf/tests/adaptive_smoothing_wkf_test.py:
```diff
@@ -0,0 +1,114 @@
+        computed = pd.read_csv(fname_out)
+        np.testing.assert_almost_equal(expected, computed['nocc'], decimal=4)
+    
+#    def test_adap_smoothing_wkf(self):
```
Do you still need this code commented out?

kirstybayliss replied:
No, I don't believe that I do! Have removed.

mmpagani reviewed (approved):
Thanks a lot for preparing this and sorry for the delay of my review. Apart from a very minor comment, it looks good to me.

kirstybayliss commented:
Since December:

-  Added normalisation so that event count output is easily and directly comparable to Gaussian outputs. Unnormalised outputs can still be returned by the original function if maxdist is not supplied, but wkf method will return normalised values by default. Updated test results accordingly (only to normalise values where required).
-  Moved information gain function so it can be called outside of the adaptive smoothing and can be more easily used with Gaussian smoothing outputs
- Updated information gain to avoid nans when there are zero events forecast in a cell
- Fixed a bug I introduced while making these changes
